### DISCUSSION TITLE
Interaction on large touch screens

### DISCUSSION THREAD

**Discussion by forbeshawkins** [google/model-viewer#2520] [Q&A]
Hi folks,

Touchmove events are (by design I think) not processed until you have dragged across a certain number of pixels. This is, I assume, to help differentiate between a tap where the finger might moves slightly, and a deliberate move event.

This effect is exaggerated on larger screens. It means that when you look at the model-viewer on a very large touch-screen, as you begin dragging to rotate, the model doesn't move until you have dragged some distance. Once the threshold is reached, you get this sudden jump as cached touchmove events are processed.  On a 55 inch screen, I need to drag a couple of cm before the model jumps and then starts responding. Feels kind of glitchy

Is this what they call touch-slop? Seems to be implemented in browsers (Firefox, chromium based) - the same problem happens if I touch-scroll a web page. It's certainly not a hardware level thing - touch dragging responds as expected in other apps. With model-viewer, there also seems to be some touch event implementation using three - not sure how it relates to the browser implementation.

Anyway before I dig much further, am wondering if anyone else has tripped up on this and found a workaround?

thanks!

**Comment by elalish** [comment]
That's interesting and I have not seen this. I can't repro on my chromebook tablet. I don't think that's how touch events work, but I'm not completely sure. You might also check your graphics performance; sometimes screens with a lot of pixels have trouble keeping up because of that.

**Reply by forbeshawkins** [reply]
Thanks for helping out @elalish. I am basing my assumption on a few things I've read, such as:

https://docs.google.com/document/d/12k_LL_Ot9GjF8zGWP9eI_3IMbSizD72susba0frg44Y/mobilebasic
"chrome currently suppresses touchmove events during the slop region, so click will NEVER involve a touchmove"

and

https://www.chromium.org/user-experience/touch-access
"Swipe Gestures... A gesture will be registered if... the distance moved is far enough (outside the "slop region") to determine that the movement was deliberate"

As I say, I see this on a 55 inch large screen - maybe because the pixel radius is so large? I don't see it on smaller screens,  so wouldn't expect to see it on a tablet (unless it's  connected it to a large display?)

Also, once the touchmove events start registering, it works beautifully so I suspect it isn't related to performance. Perhaps I can post a video here to better illustrate...

**Comment by forbeshawkins** [comment]
Not sure how well you can see from this. I have to drag my finger a distance before the model starts moving. But after that it responds really well...

https://user-images.githubusercontent.com/1035696/123891030-9a7d6700-d99b-11eb-9a38-acf977f65cb2.mp4

**Reply by dubi** [reply]
hi @forbeshawkins do you still see this issue even with the latest version ?
Thanks
